src/main/java/at/hannibal2/skyhanni/features/misc/MagicalPowerDisplay.kt

```diff
@@ -0,0 +1,128 @@
+    * REGEX-TEST: RARE ACCESSORY
+    * REGEX-TEST: EPIC DUNGEON ACCESSORY
+    * REGEX-TEST: a LEGENDARY ACCESSORY a
+    * */
+    private val accessoryLorePattern by RepoPattern.pattern(
+        "accessory.lore",
+        """a?\s*(COMMON|UNCOMMON|RARE|EPIC|LEGENDARY|MYTHIC)\s*(?:DUNGEON\s*)?ACCESSORY\s*a?"""
+    )
+
+    @SubscribeEvent
+    fun onRenderItemTip(event: RenderItemTipEvent) {
+        if (!isEnabled()) return
+        if (!acceptedInvPattern.matches(InventoryUtils.openInventoryName().stripControlCodes())) return
+
+        val item = event.stack
+        val rarity = item.isAccessory() ?: return
+        val itemID = item.getInternalNameOrNull() ?: return
+
+        var endMP = rarity.toMP() ?: run {
+            ErrorManager.skyHanniError(
+                "Unknown rarity '$rarity' for item '${item.displayName}§7'"
+            )
+        }
+        if (itemID == "HEGEMONY_ARTIFACT".asInternalName())
+            endMP *= 2
+        if (itemID == "RIFT_PRISM".asInternalName())
+            endMP = 11
+        if (item.isAbicase())
+            endMP += contactAmount / 2
+        event.stackTip = "${if (config.colored) rarity.chatColorCode else "§7"}${endMP}"
+    }
+
+    @SubscribeEvent
+    fun onInventoryOpened(event: InventoryOpenEvent) {
+        if (!isEnabled()) return
+        if (event.inventoryName.startsWith("Abiphone")) {
+            val theBookLore = event.inventoryItems[51]?.getLore() ?: return
+            ChatUtils.debug("Checking Abiphone for contacts")
+            for (line in theBookLore) {
+                val stripped = line.stripControlCodes()
+                if (stripped.startsWith("Your contacts: ")) {
+                    contactAmount = stripped.split(" ")[2].split("/")[0].toInt()
+                    return
+                }
+            }
+        }
+    }
+
+    private fun ItemStack.isAbicase(): Boolean {
+        val id = this.getInternalNameOrNull() ?: return false
+        return id.startsWith("ABICASE_")
+    }
+
+    private fun LorenzRarity.toMP(): Int? {
+        return when (this) {
+            LorenzRarity.COMMON, LorenzRarity.SPECIAL -> 3
+            LorenzRarity.UNCOMMON, LorenzRarity.VERY_SPECIAL -> 5
+            LorenzRarity.RARE -> 8
+            LorenzRarity.EPIC -> 12
+            LorenzRarity.LEGENDARY -> 16
+            LorenzRarity.MYTHIC -> 22
+            else -> null
+        }
+    }
+
+    private fun ItemStack.isAccessory(): LorenzRarity? {
+        val lore = this.getLore()
+        for (line in lore) {
+            val stripped = line.stripControlCodes()
+
+            if (stripped == "SPECIAL HATCESSORY") {
+                return LorenzRarity.SPECIAL
+            } else if (stripped == "a VERY SPECIAL HATCESSORY a") {
+                return LorenzRarity.VERY_SPECIAL
+            }
+
+            if (accessoryLorePattern.matches(stripped)) {
+                return this.getItemRarityOrNull()
+            }
+        }
+        return null
+    }
```

Comment:
We have ItemCategory for that,
Simply use 
'''suggestion
private fun ItemStack.isAccessory(): LorenzRarity?{
   if(this.itemCategoryOrNull() != ItemCategory.ACCESSORY && this.itemCategoryOrNull() != ItemCategory.HATCESSORY) return null
   return this.getItemRarityOrNull()
}
'''
The syntax might be a bit off. I did just type that out of my head without checking if the names where correct.
If the itemrarity does not work for special and very special let me know. Since that should be fixed in the rarity detection code.